cmake/helpers.cmake

```diff
@@ -104,3 +108,46 @@ function(add_umf_library)
     add_umf_target_compile_options(${ARG_NAME})
     add_umf_target_link_options(${ARG_NAME})
 endfunction()
+
+# Add sanitizer ${flag}, if it is supported, for both C and C++ compiler
+macro(add_sanitizer_flag flag)
+    # Save current 'CMAKE_REQUIRED_LIBRARIES' state and temporarily extend it with
+    # '-fsanitize=${flag}'. It is required by CMake to check the compiler for
+    # availability of provided sanitizer ${flag}.
+    if(WINDOWS)
+        set(SANITIZER_FLAG "/fsanitize=${flag}")
+        set(SANITIZER_ARGS "")
+    else()
+        set(SANITIZER_FLAG "-fsanitize=${flag}")
+        set(SANITIZER_ARGS "-fno-sanitize-recover=all")
+    endif()
+
+	set(SAVED_CMAKE_REQUIRED_LIBRARIES ${CMAKE_REQUIRED_LIBRARIES})
+	set(CMAKE_REQUIRED_LIBRARIES "${CMAKE_REQUIRED_LIBRARIES} ${SANITIZER_FLAG}")
+
+    if(${flag} STREQUAL "address")
+        set(check_name "HAS_ASAN")
+    elseif(${flag} STREQUAL "undefined")
+        set(check_name "HAS_UBSAN")
+    elseif(${flag} STREQUAL "thread")
+        set(check_name "HAS_TSAN")
+    endif()
+
+    # Check C and CXX compilers for given sanitizer flag.
+	check_c_compiler_flag("${SANITIZER_FLAG}" "C_${check_name}")
+	check_cxx_compiler_flag("${SANITIZER_FLAG}" "CXX_${check_name}")
+	if (${C_${check_name}} OR ${CXX_${check_name}})
+        # Set appropriate linker flags for building executables.
+		set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${SANITIZER_FLAG} ${SANITIZER_ARGS}")
```

Comment:
not sure about this one, but shouldn't we re-set this `CMAKE_EXE_LINKER_FLAGS` after the test?

Reply:
I assume that we would want every executable built in UMF to be sanitized. What would re-setting this variable achieve?

Reply:
yeah, sorry, mea culpa, I mistakenly thought it should be reset like the `SAVED_...` variable. It's all fine